pages/Signup.tsx

```diff
@@ -19,12 +23,91 @@ const Signup: NextPage = () => {
         });
     };
 
+    const formSubmitHandler = (event: React.FormEvent<EventTarget>): void => {
+        event.preventDefault();
+
+        const { username, email, password } = formState;
+
+        // test for fetch request to API handler in pages/api/hello.ts
+        try {
+            fetch('/api/hello', {
+              method: 'POST',
+              body: JSON.stringify({
+                  username: username,
+                  email: email,
+                  password: password
+              }),
+            })
+            .then(res => {
+              res.json().then(data => console.log(data))
+            });
+        }
+        catch(err) {
+            console.error(err);
+        }
+        finally {
+            setFormState({
+                username: '',
+                email: '',
+                password: ''
+            });
+
+            router.reload();
+            // redirect to Home once set up
+            // router.replace('/home');
+        }
+    };
+
+    const loginHandler = (event: React.FormEvent<EventTarget>): void => {
+        event.preventDefault();
+
+        const { email, password } = formState;
+
+        // test for fetch request to API handler in pages/api/hello.ts
+        try {
+            fetch('/api/hello', {
+              method: 'POST',
+              body: JSON.stringify({
+                  email: email,
+                  password: password
+              }),
+            })
+            .then(res => {
+              res.json().then(data => console.log(data))
+            });
+        }
+        catch(err) {
+            console.error(err);
+        }
+        finally {
+            setFormState({
+                username: '',
+                email: '',
+                password: ''
+            });
+
+            router.reload();
+            // redirect to Home once set up
+            // router.replace('/home');
+        }
+    };
```

Comment:
This is really just a question, because I don't use `fetch` very often, but is there an advantage to having the exterior `try/catch` as opposed to just chaining the `then/catch/finally` to the `fetch`?

For example:

```suggestion
        fetch('/api/hello', {
            method: 'POST',
            body: JSON.stringify({
                email: email,
                password: password
            }),
        })
            .then((res) => res.json() )
            .then(data => console.log(data))
            .catch((err) => console.error(err))
            .finally(() => {
                setFormState({
                    username: '',
                    email: '',
                    password: ''
                })
                router.reload();
                // redirect to Home once set up
                // router.replace('/home');
            });
    };
```